packages/prediction-app/src/utils/useApiDataQuery.ts

```diff
@@ -0,0 +1,42 @@
+import { useDataEngine } from "@dhis2/app-runtime"
+import {
+    useQuery,
+    QueryFunction,
+    UseQueryOptions,
+    QueryKey,
+} from "@tanstack/react-query";
+import { ResourceQuery } from '../interfaces/apiQueryTypes';
+
+
+type UseApiDataQueryProps<
+    TResultData,
+    TError = Error,
+    TData = TResultData,
+    TQueryKey extends QueryKey = QueryKey,
+> = Omit<UseQueryOptions<TResultData, TError, TData, TQueryKey>, 'queryFn'> & {
+    query: ResourceQuery;
+};
+
+export const useApiDataQuery = <
+    TResultData,
+    TError = Error,
+    TData = TResultData,
+    TQueryKey extends QueryKey = QueryKey,
+>({
+    query,
+    queryKey,
```

Comment:
I've had some success by using the query as the query-key. Right now the `queryKey` are created ad-hoc and don't really reference each other, which can be hard to track/keep consistent down the line.

Reply:
Discussed this async on Slack. Decided to keep our custom query keys for now, but open for discussion at a later point.